caas/kubernetes/provider/application/application.go

```diff
@@ -2057,6 +2083,18 @@ func (a *app) configureStorage(
 				return errors.Trace(err)
 			}
 		}
+
+		if unitAttachments, ok := filesystemUnitAttachments[fs.StorageName]; ok {
+			for _, attachment := range unitAttachments {
+				// Convert unit-{appName}-{unitNumber} to {appName}-{storageName}-{uniqueId}-{appName}-{unitNumber}.
```

Comment:
Using unit name in KubernetesFilesystemUnitAttachmentParams will result in a change here too